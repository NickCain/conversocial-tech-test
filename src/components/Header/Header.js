import React, { Component } from 'react';

import './Header.css';

class Header extends Component {
  render() {
    return (
      <div className="Header">
        <div className="App-header">
          <h2>An Awesome Chat App</h2>
        </div>
      </div>
    );
  }
}

export default Header;
