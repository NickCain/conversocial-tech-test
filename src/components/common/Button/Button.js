import React, { Component } from 'react';

import './Button.css';

class Button extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const { text, onClick } = this.props;
    return (
      <button className="Button" onClick={onClick}>
        {text}
      </button>
    );
  }
}

export default Button;
