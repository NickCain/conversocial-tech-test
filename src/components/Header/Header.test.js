import { mount, shallow } from 'enzyme';
import React from 'react';
import Header from './Header.js'

import Button from '../../components/common/Button/Button';

let wrapper;
let shallowWrapper;

describe('Header - Component', () => {
  describe('Header renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <Header />
      );
      shallowWrapper = shallow(
        <Header />
      );
    });

    it('should render the Header component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });

    it('should render text header', () => {
      expect(wrapper.text()).toContain('An Awesome Chat App');
    });

    it('should render one <Button /> component', () => {
      expect(shallowWrapper.find(Button)).toHaveLength(1);
    });
  });
});
