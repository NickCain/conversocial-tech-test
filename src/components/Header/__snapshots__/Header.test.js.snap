// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header - Component Header renders should render the Header component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Header"
>
    <div
        className="App-header"
    >
        <h2>
            An Awesome Chat App
        </h2>
        <div
            className="Button-wrapper"
        >
            <Button
                text="Add Chat"
            />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="Header"
>
      <div
            className="App-header"
      >
            <h2>
                  An Awesome Chat App
            </h2>
            <div
                  className="Button-wrapper"
            >
                  <Button
                        text="Add Chat"
                  />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header />,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="Header"
>
          <div
                    className="App-header"
          >
                    <h2>
                              An Awesome Chat App
                    </h2>
                    <div
                              className="Button-wrapper"
                    >
                              <Button
                                        text="Add Chat"
                              />
                    </div>
          </div>
</div>,
        "_debugID": 10,
        "_renderedOutput": <div
          className="Header"
>
          <div
                    className="App-header"
          >
                    <h2>
                              An Awesome Chat App
                    </h2>
                    <div
                              className="Button-wrapper"
                    >
                              <Button
                                        text="Add Chat"
                              />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header />,
}
`;
